File: catalog-api/enumerate-products-python/enumerate-products.py
#!/usr/bin/env python

import boto3

client = boto3.client('marketplace-catalog', region_name='us-east-1')

catalog_name = 'AWSMarketplace'
entity_types = ['DataProduct']

for entity_type in entity_types:
  print("Enumerating " + entity_type)
  try:
    res = client.list_entities(
      Catalog=catalog_name, 
      EntityType=entity_type
    )

    for entity_summary in res['EntitySummaryList']:
      print(' ' + entity_summary['Name'])

  except (client.exceptions.ValidationException, client.exceptions.AccessDeniedException):
    print(" skipping, unauthorized")
    pass

File: catalog-api/enumerate-products-ruby/enumerate-products.rb
# frozen_string_literal: true

require 'aws-sdk-marketplacecatalog'

Aws.config.update(
  region: ENV['AWS_REGION'] || 'us-east-1',
  credentials: Aws::Credentials.new(
    ENV['AWS_ACCESS_KEY_ID'],
    ENV['AWS_SECRET_ACCESS_KEY'],
    ENV['AWS_SESSION_TOKEN']
  )
)

catalog_name = 'AWSMarketplace'

catalog = Aws::MarketplaceCatalog::Client.new

product_types = ['DataProduct']

product_types.each do |product_type|
  begin
    puts "Enumerating your #{product_type}s ..."

    entities = catalog.list_entities(
      catalog: catalog_name,
      entity_type: product_type,
    ).entity_summary_list

    entities.each do |entity|
      puts " #{entity.entity_id}: #{entity.name}"

      # entity details
      # described_entity = catalog.describe_entity(catalog: catalog_name, entity_id: entity.entity_id)
      # described_entity_details = JSON.parse(described_entity.details)
    end
  rescue Aws::MarketplaceCatalog::Errors::ServiceError => e
    puts " skipping, #{e.message}"
  end
end

puts 'Done.'
